File: src/main/java/com/manageway/domain/user/records/Phone.java
package com.manageway.domain.user.records;

import com.manageway.StaticUtils;
import org.springframework.util.Assert;

public record Phone(String value) {
    public Phone {
        StaticUtils.validateNotNullAndNotEmpty(value, "Phone must not be null or empty");

        String validRegex = "^\\+?[0-9\\-\\s]+$";
        Assert.isTrue(value.matches(validRegex),
                String.format("Phone must be valid, but was %s", value));
    }
}

File: src/main/kotlin/com/manageway/domain/user/Phone.kt
package com.manageway.domain.user

data class Phone(
    val value: String
) {
    init {
        require(value.length in 5..20) { "Phone must be between 5 and 20 characters" }
        require(value.matches(Regex("^[0-9]+\$"))) { "Phone must be valid" }
    }
}
